apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Snippets/DataModelSnippets.cs

```diff
@@ -273,6 +273,78 @@ public CustomCity FromFirestore(object value)
         }
         // End sample
 
+        // Sample: ComplexConverterIgnoresNull
+        [FirestoreData(ConverterType = typeof(CustomPersonConverter))]
+        public class CustomPerson
+        {
+            // Note: no [FirestoreProperty] attributes. The converter
+            // is doing all the work.
+            public string FirstName { get; }
+            public string MiddleName { get; }
+            public string LastName { get; }
+
+            public CustomPerson(string firstName, string middleName, string lastName)
+            {
+                FirstName = firstName;
+                MiddleName = middleName;
+                LastName = lastName;
+            }
+        }
+
+        public class CustomPersonConverter : IFirestoreConverter<CustomPerson>
+        {
+            public object ToFirestore(CustomPerson value)
+            {
+                IDictionary<string, object> map = new Dictionary<string, object>
+                {
+                    { "FirstName", value.FirstName },
+                    { "LastName", value.LastName },
+                };
+                if (value.MiddleName != null)
+                {
+                    map.Add("MiddleName", value.MiddleName);
+                }
+                return map;
+            }
+
+            public CustomPerson FromFirestore(object value)
+            {
+                if (value is IDictionary<string, object> map)
+                {
+                    // Any exceptions thrown here will be propagated directly. You may wish to be more
+                    // careful, if you need to control the exact exception thrown used when the
+                    // data is incomplete or has the wrong type.
+                    return new CustomPerson(
```

Comment:
I think it would be worth extracting these out to locals first. I'd probably use `TryGetValue` rather than `ContainsKey` followed by the indexer, too.